MapsIndoorsSamples/app/src/main/java/com/mapspeople/mapsindoorssamples/ui/reverseGeoCode/ReverseGeoCodeFragment.kt

```diff
@@ -0,0 +1,102 @@
+import com.mapsindoors.core.MPPoint
+import com.mapsindoors.core.MapControl
+import com.mapsindoors.core.MapsIndoors
+import com.mapsindoors.core.errors.MIError
+import com.mapsindoors.googlemaps.MPMapConfig
+import com.mapsindoors.googlemaps.converters.LatLngBoundsConverter
+import com.mapspeople.mapsindoorssamples.R
+import com.mapspeople.mapsindoorssamples.databinding.FragmentReverseGeoCodeBinding
+
+class ReverseGeoCodeFragment : Fragment(), OnMapReadyCallback{
+    private var _binding: FragmentReverseGeoCodeBinding? = null
+
+    private val binding get() = _binding!!
+
+    private var mMap: GoogleMap? = null
+    private var mMapView: View? = null
+    private var mMapControl: MapControl? = null
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        _binding = FragmentReverseGeoCodeBinding.inflate(inflater, container, false)
+
+        MapsIndoors.load(requireActivity().applicationContext, "57e4e4992e74800ef8b69718", null)
+
+        val root: View = binding.root
+        val supportMapFragment = childFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
+        mMapView = supportMapFragment.view
+        supportMapFragment.getMapAsync(this)
+        return root
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onMapReady(map: GoogleMap) {
+        mMap = map
+        if (mMapView != null) {
+            initMapControl(mMapView!!)
+        }
+    }
+
+    private fun initMapControl(view: View) {
+        val mapConfig: MPMapConfig = MPMapConfig.Builder(requireActivity(), mMap!!, getString(R.string.google_maps_key), view, true).build()
+        //Creates a new instance of MapControl
+        MapControl.create(mapConfig) { mapControl: MapControl?, miError: MIError? ->
+            mMapControl = mapControl
+            //Enable Live Data on the map
+            if (miError == null) {
+                //No errors so getting the first venue (in the white house solution the only one)
+                val venue = MapsIndoors.getVenues()?.defaultVenue
+                activity?.runOnUiThread {
+                    if (venue != null) {
+                        //Animates the camera to fit the new venue
+                        mMap!!.animateCamera(
+                            CameraUpdateFactory.newLatLngBounds(
+                                LatLngBoundsConverter.toLatLngBounds(venue.bounds!!),
+                                19
+                            )
+                        )
+                    }
+                }
+            }
+
+            setupListener()
+        }
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun setupListener() {
+        mMapControl?.let { mapControl ->
+            //Use a MapClick Listener to get a coordinate to perform the reverse geocode with.
+            mapControl.setOnMapClickListener { mpLatLng, mpLocations ->
```

Comment:
If you want to explicitly ignore the locations returned, replace the `mpLocations` with `_`

Reply:
As this is for documentation purpose, i like that the users know what the listener returns (espescially in the code examples), and is it is referenced in the comment below i think it is best to keep it commented, even if our linter disapproves